apps/passport-server/src/database/queries/e2ee.ts

```diff
@@ -38,6 +38,37 @@ export async function insertEncryptedStorage(
   );
 }
 
+/**
+ * Transactionally deletes row at at the old blob key,
+ * upserts the encrypted data stored at the new blob key,
+ * and then updates the user's salt.
+ */
+export async function updateEncryptedStorage(
+  dbPool: Pool,
+  oldBlobKey: string,
+  newBlobKey: string,
+  uuid: string,
+  newSalt: string,
+  encryptedBlob: string
+): Promise<void> {
+  await sqlQuery(dbPool, "BEGIN;");
+  await sqlQuery(dbPool, "DELETE FROM e2ee WHERE blob_key = $1;", [oldBlobKey]);
+  await sqlQuery(
+    dbPool,
+    `INSERT INTO e2ee(blob_key, encrypted_blob) VALUES
+      ($1, $2) ON CONFLICT(blob_key) DO UPDATE SET encrypted_blob = $1;`,
+    [newBlobKey, encryptedBlob]
+  );
+  await sqlQuery(
+    dbPool,
+    `UPDATE commitments
+    SET salt = $2
+    WHERE uuid = $1;`,
+    [uuid, newSalt]
+  );
+  await sqlQuery(dbPool, "COMMIT;");
```

Comment:
this seems pretty fragile - what if another `sqlQuery` gets called between one of these two? I don't really know how to model that behavior. can you look into how to do transactions properly in the `pg` library?

Reply:
I think the key is to grab an explicit Client from the pool to keep all your statements together, like the example here: https://node-postgres.com/features/transactions